### DISCUSSION TITLE
Upgrading flexihal firmware

### DISCUSSION THREAD

**Discussion by jpraus** [Expatria-Technologies/Flexi-HAL#8] [Q&A]
Sorry for dumb question, I just received a flexiHal board and wanted to upgrade the firmware for printnc to the latest version. Not 100% sure what are the steps. I've connected the flexiHal with USB and pressed RST twice. USB driver appeared on my laptop. So I've copied one of the uf2 file from **Stable Release for 0.9.9.8** (FLEXI_HAL_PRINTNC_20231229_180124.1.uf2). But as soon as I copy it over to the USB drive the drive gets disconnected. Not sure if the firmware gets updated like this...

This is my info_uf2.txt file:
```
TinyUF2 Bootloader 0.9.4-5-gb309781-dirty - st/cmsis_device_f4 (v2.6.4-1-g2615e86) st/stm32f4xx_hal_driver (v1.7.9) tinyusb (0.12.0-203-ga4cfd1c69)
Model: ST STM32F446 Flexi-Hal
Board-ID: STM32F446-Flexi-Hal
Date: Jun 17 2022
```

**Comment by andrewmarles** [comment]
yes, that is how it works.  The UF2 bootloader is a simple port from Adafruit's UF2 implementation.  You can read some more details about how it works here:

https://learn.adafruit.com/adafruit-feather-m0-express-designed-for-circuit-python-circuitpython/uf2-bootloader-details#using-the-mass-storage-bootloader-2929754
